runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1532,33 +1847,95 @@ TR_J9InlinerPolicy::createUnsafeGetWithOffset(TR::ResolvedMethodSymbol *calleeSy
          TR::Node::createWithSymRef(storeOpCode, 1, 1, unsafeCallWithConversion, newTemp);
       }
 
+   // If we can get a J9Class or we need conversion we generate test to branch to direct access if low bit is not tagged
+   // Else in case we get NULL instead of j9Class we generate test to branch to indirect access if low bit is tagged
+   TR::TreeTop *lowTagCmpTree = genClassCheckForUnsafeGetPut(offset, needNotLowTagged);
+
    TR::TreeTop* directAccessTreeTop =
       genDirectAccessCodeForUnsafeGetPut(callNodeTreeTop->getNode(), false, true);
-   TR::TreeTop* arrayDirectAccessTreeTop = conversionNeeded
-      ? genDirectAccessCodeForUnsafeGetPut(callNodeWithConversion, conversionNeeded, true)
-      : NULL;
+
+   // If we need the array block but not the array check, we need to replace the directAccessBlock with the arrayDirectAccessBlock.
+   // Thus, in this scenario, rather than generate an additional block as the arrayDirectAccessBlock, we will just directly modify
+   // the directAccessBlock.
+   TR::TreeTop* arrayDirectAccessTreeTop = NULL;
+
```

Comment:
There are 8 possible combinations among these 3 variables: conversionNeeded, arrayCheckNeeded, and arrayBlockNeeded. are you sure you covered them all with 3 prongs (if and else) below?

I guessed true-arrayCheckNeeded  implies true-arrayBlockNeeded, also true-conversionNeeded implies true-arrayBlockNeeded.  that reduced to 6 combinations.  Just point it out for you to double-check here if you got them all covered. At least, you can comment in code to that effect.

Reply:
when i read more of the code, the comment above looks like a general comment applicable to other places in code too.  you are doing if-then-else-if-then-else to generate different ILs.  it should be described to the effect that every possible case is covered.